tools/codegen/dest/lazy_ir.py

```diff
@@ -174,37 +229,40 @@ def this_shape(i: int) -> str:
                 shapes_str = ','.join([this_shape(i) for i in range(returns_length)])
                 meta_out = "std::vector<Shape> shapes{" + shapes_str + "};"
 
+            # TODO: INTEGRATION POINT HERE:
             meta_str = f"""auto out_meta = at::meta::{schema.aten_name}({', '.join(str(t.name) for t in all_types)});
         {meta_out}"""
         else:
-            shape_sig = ComputeShapeSignature(func)
+            shape_sig = ComputeShapeSignature(metadata.kernel, func)
             meta_str = f"""
         auto shapes = {shape_sig.shape_call};"""
+
         meta_str += f"""
         TORCH_INTERNAL_ASSERT(shapes.size() == {returns_length});"""
 
         node_str = f"""auto node = torch::lazy::MakeNode<ir::ops::{schema.node_name}>({node_ctor_input_str},
                                                                                       std::move(shapes));"""
+        first_tensor_name = value_types_names[0]
+        bridge_str = """auto result = torch::lazy::CreateAtenFromLtcTensor(
+                torch::lazy::LazyTensor::Create(std::move(node), *common_device));"""
 
-        assert len(value_types) > 0, f"Only supporting tensor ops so far, none found in {sig}"
-        first_tensor = value_types[0]
-        bridge_str = f"""auto result = CreateAtenFromLtcTensor(lazy_{first_tensor.name}.CreateFrom(node));"""
         if returns_length > 1:
             bridge_str = f"""std::vector<{self.tensor_class}> lazy_tensors;
         for (int i = 0; i < {returns_length}; i++) {{
-            lazy_tensors.push_back(lazy_{first_tensor.name}.CreateFrom(torch::lazy::Value(node, i)));
+            lazy_tensors.push_back(torch::lazy::LazyTensor::Create(torch::lazy::Value(node, i), *common_device));
         }}
-        auto result = TupleAtenFromLtcTensors<{returns_length}>(lazy_tensors);"""
-        if schema.name.name.inplace:
+        auto result = torch::lazy::TupleAtenFromLtcTensors<{returns_length}>(lazy_tensors);"""
+
+        if schema.name.name.inplace or func.func.is_out_fn():
             assert returns_length == 1, "We assumed there was no such case where an op is an in-place variant " \
                                         "and has tuple outputs."
-            bridge_str = f"""lazy_{first_tensor.name}.SetInPlaceIrValue(node);
-        auto& result = {first_tensor.name};"""
+            bridge_str = f"""lazy_{first_tensor_name}.SetInPlaceIrValue(node);
+        auto& result = {first_tensor_name};"""
 
 
         return [f"""\
-    // TODO(alanwaketan): Quite a lot inefficient copy-by-value there. Let's optimize it.
-    {sig.decl(name=f"{self.class_method_name}::{schema.aten_name}")} {{
+    {sig.decl(name=f"{self.class_method_name}::{metadata.kernel}")} {{
+        {fallback_str}
```

Comment:
Just confirming - so this PR also adds "per-op-defined eager fallback"?

Reply:
the fallback thing here is just a debugging feature.  It lets you set an env variable that will force a particular operator to fall back.